libpod/image/image.go

```diff
@@ -182,6 +182,16 @@ func (i *Image) getLocalImage() (*storage.Image, error) {
 	if i.InputName == "" {
 		return nil, errors.Errorf("input name is blank")
 	}
+	// Check if the input name has a transport and if so strip it
+	dest, err := alltransports.ParseImageName(i.InputName)
+	if err != nil {
+		return nil, errors.Wrapf(err, imageError)
+	}
+
+	if i.InputName != dest.DockerReference().String() {
+		i.InputName = dest.DockerReference().String()
```

Comment:
could the if be dropped in favor of always doing the assignment?

Reply:
it could but I worry about other kinds of inputs going through that parse and giving me unexpected results ...

Reply:
isn't InputName a string? My Go-Fu is not the strongest but I think that if the string are the same (and now you'd skip the assignment), then assigning one to the other will not make any difference and you avoid a comparison.

Reply:
this condition should be ... if the users input name doesn't equal the parsed reference, then likely there is a transport name in the input ... i think we are ok?

Reply:
yes but the end result is that `i.InputName = dest.DockerReference().String()` no matter what was the initial value for `i.InputName`, so you could just assign directly to it.  In any case, your call :slightly_smiling_face: